src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java

```diff
@@ -2394,6 +2399,48 @@ public void shouldSetCodicilsNumberNullWhenWillHasCodicilsNo() {
         assertEquals(null, callbackResponse.getData().getWillNumberOfCodicils());
     }
 
+    @Test
+    public void shouldSetDeceasedDeathCertificateNull() {
+        caseDataBuilder.deceasedDeathCertificate(DECEASED_DEATH_CERTIFICATE);
+        caseDataBuilder.deceasedDiedEngOrWales(NO);
+
+        when(callbackRequestMock.getCaseDetails()).thenReturn(caseDetailsMock);
+        when(caseDetailsMock.getData()).thenReturn(caseDataBuilder.build());
+
+        CallbackResponse callbackResponse = underTest.transformCase(callbackRequestMock);
+
+        assertEquals(null, callbackResponse.getData().getDeceasedDeathCertificate());
+    }
+
+    @Test
+    public void shouldSetDeceasedForeignDeathCertTranslationNull() {
+        caseDataBuilder.deceasedForeignDeathCertTranslation(YES);
+        caseDataBuilder.deceasedForeignDeathCertInEnglish(YES);
+        caseDataBuilder.deceasedDiedEngOrWales(NO);
+
+        when(callbackRequestMock.getCaseDetails()).thenReturn(caseDetailsMock);
+        when(caseDetailsMock.getData()).thenReturn(caseDataBuilder.build());
+
+        CallbackResponse callbackResponse = underTest.transformCase(callbackRequestMock);
+
+        assertEquals(null, callbackResponse.getData().getDeceasedForeignDeathCertTranslation());
+    }
+
+    @Test
+    public void shouldSetDeceasedEnglishForeignDeathCertANDForeignDeathCertTranslationNull() {
+        caseDataBuilder.deceasedForeignDeathCertInEnglish(NO);
+        caseDataBuilder.deceasedForeignDeathCertTranslation(YES);
+        caseDataBuilder.deceasedDiedEngOrWales(YES);
+
+        when(callbackRequestMock.getCaseDetails()).thenReturn(caseDetailsMock);
+        when(caseDetailsMock.getData()).thenReturn(caseDataBuilder.build());
+
+        CallbackResponse callbackResponse = underTest.transformCase(callbackRequestMock);
+
+        assertEquals(null, callbackResponse.getData().getDeceasedForeignDeathCertInEnglish());
+        assertEquals(null, callbackResponse.getData().getDeceasedForeignDeathCertTranslation());
+    }
+
     @Test
     public void shouldSetSOT() {
         caseDataBuilder.applicationType(ApplicationType.PERSONAL);
```

Comment:
got e2e tests to update as well but prob best to wait for my pr to get those ship-shape first